code/modules/station_goals/meteor_shield.dm

```diff
@@ -109,6 +120,9 @@
 	if(obj_flags & EMAGGED)
 		update_emagged_meteor_sat(user)
 
+	var/datum/station_goal/station_shield/goal = SSstation.get_station_goal(__IMPLIED_TYPE__)
+	goal?.update_coverage()
```

Comment:
is this safe? we base it off view and that can change can't it?

Reply:
We cannot reliably check that the view has changed in any way that isn't super expensive or super shitcode

Reply:
right so isn't this fucking up behavior then?

Reply:
Nope, the original proc only ever got used with the console, and SSticker calls update_coverage again before checking completion

Reply:
handled 
![image](https://github.com/tgstation/tgstation/assets/58055496/e898209c-db19-4db6-9ebe-fd813f04118e)